src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -408,9 +408,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <When Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_IsExecutable)' == 'true'">
       <PropertyGroup Condition="'$(SelfContained)' != 'true'">
         <!-- TODO: https://github.com/dotnet/sdk/issues/20 Need to get the DotNetHost path from MSBuild -->
-        <RunCommand Condition="'$(RunCommand)' == ''">dotnet</RunCommand>
+        <ExecutingRuntime Condition="'$(ExecutingRuntime)' == ''">dotnet</ExecutingRuntime>
+        <RunCommand Condition="'$(RunCommand)' == ''">$(ExecutingRuntime)</RunCommand>
+
+        <_NetCoreRunArguments>&quot;$(TargetPath)&quot;</_NetCoreRunArguments>
+        <_NetCoreRunArguments Condition="'$(ExecutingRuntime)' == 'dotnet'">exec $(_NetCoreRunArguments)</_NetCoreRunArguments>
```

Comment:
@eerhardt Is there any reason why we can't drop the exec here? I'm thinking about suggesting some things that could generalize yet simplify the code if we can.

Reply:
I believe it can be dropped. Typically `exec` is necessary when you want to pass in optional arguments into `dotnet.exe`, like `--depsfile` and `--runtimeconfig`, which we aren't passing here.

The `exec` arg was ported from the project.json implementation:  https://github.com/dotnet/cli/blob/b5d312e7faff0e551b58e493bbb1d0de48999954/src/dotnet/commands/dotnet-run/RunCommand.cs#L191